Improved handling of grantor access to organizations after takeover

eliykat opened the pull request:
This PR is required to support the changes described in https://github.com/bitwarden/server/pull/1132. Refer to that PR for explanation.

cscharf commented on src/locales/en/messages.json:
```diff
@@ -3417,6 +3417,9 @@
   "emergencyAccessDesc": {
     "message": "Grant and manage emergency access for trusted contacts. Trusted contacts may request access to either View or Takeover your account in case of a emergency. Visit our help page for more information and details into how zero knowledge sharing works."
```
While you're in here, can you please correct this one too?
> your account in case of a emergency.

Should be
> your account in case of **an** emergency.

cscharf reviewed (commented):
Some basic feedback, would love one of the other team members to take a look.

eliykat commented:
Thanks Chad! I'll wait for another review of this one before merging any of the other PRs, so they can all be merged at once.

cscharf reviewed (approved):
Actually, I think this looks good. Thanks Thomas!

cscharf reviewed (changes requested):
Actually, just noticed the build is failing. May need to take a look at that.

eliykat commented:
Except for an implicit any error (which I've just fixed), the build errors are only due to changes in jslib. Works fine if you have the updated jslib hooked up locally.

cscharf reviewed (approved):
that's better 😁

cscharf reviewed (changes requested):
nm, spoke too soon
```
ERROR in src/app/settings/emergency-access-takeover.component.ts:55:35 - error TS2339: Property 'policy' does not exist on type 'EmergencyAccessTakeoverResponse'.

55         if (this.takeoverResponse.policy != null && this.takeoverResponse.policy.length > 0)
                                     ~~~~~~
src/app/settings/emergency-access-takeover.component.ts:55:75 - error TS2339: Property 'policy' does not exist on type 'EmergencyAccessTakeoverResponse'.

55         if (this.takeoverResponse.policy != null && this.takeoverResponse.policy.length > 0)
                                                                             ~~~~~~
src/app/settings/emergency-access-takeover.component.ts:57:50 - error TS2339: Property 'policy' does not exist on type 'EmergencyAccessTakeoverResponse'.

57             const policy = this.takeoverResponse.policy.map((policyResponse: PolicyResponse) => new Policy(new PolicyData(policyResponse)));
                                                    ~~~~~~
```

eliykat commented:
The `policy` property of `EmergencyAccessTakeoverResponse` is added by changes to jslib here: https://github.com/bitwarden/jslib/pull/260/files

eliykat commented:
There you go 🥳

Hinton commented on src/app/send/send.component.html:
```diff
@@ -107,7 +107,7 @@ <h1>
                                     </a>
                                     <a class="dropdown-item" href="#" appStopClick (click)="removePassword(s)"
-                                        *ngIf="s.password">
+                                        *ngIf="s.password && !disableSend">
```
Not sure whats happening here. It seems to have picked up a unrelated change? https://github.com/bitwarden/web/blob/master/src/app/send/send.component.html#L110 seems to already have this change.

cscharf replied:
merge resolution more than likely, should be okay.

Hinton commented on src/app/settings/emergency-access-takeover.component.ts:
```diff
@@ -45,25 +50,29 @@ export class EmergencyAccessTakeoverComponent extends ChangePasswordComponent im
         }
 
-        const key = await this.cryptoService.makeKey(this.masterPassword, this.email, takeoverResponse.kdf, takeoverResponse.kdfIterations);
+        const key = await this.cryptoService.makeKey(this.masterPassword, this.email, this.takeoverResponse.kdf, this.takeoverResponse.kdfIterations);
```
Do we know what happens if the user we takeover managed to change their encryption key between the modal opening and the submit event being triggered? The modal could potentially be opened quite a bit longer than before.

cscharf replied:
That's a good question Oscar.... @eliykat ?

eliykat replied:
Good question! I just tested this situation and it didn't stop the takeover from going through or cause any unexpected behaviour. That said, I want to make sure I'm understanding the crypto-magic behind takeover before waving this through, so I'll have a chat with Oscar.

If it is a problem - or if we're just not comfortable with having stale `takeoverResponse` data client-side as a matter of best practice - the alternative would be to separate out the api calls, so that we get the grantor policies on `ngOn Init()` and move the `takeoverResponse` call back into `submit()`.

eliykat replied:
Update: it's not really an issue if the grantor changes their master password while this modal is open, but if they rotate their encryption key, it has some... extremely bad outcomes. I have successfully nuked one (1) local testing vault to verify this.

I'll separate out the api calls as I mentioned above.

eliykat commented:
I've refactored this so we get the grantor's policies via a separate api call instead of messing with the `takeoverResponse` object. More details:
* I rebased a couple of times to revert earlier commits on this branch, and to avoid pulling in unrelated commits on master (like the one that Oscar flagged). Should ultimately make the history a little tidier.
* The latest changes that need reviewing are to `emergency-access-takeover.component.ts`
* server repo has a fresh PR here: TBA
* jslib repo has a fresh PR here: TBA
